### PR TITLE
System Tests: Improvement for wait method

### PR DESCRIPTION
**Pull request by serrss**
### Type of change
- Enhancement

### Description
 Improved the logic for the method waitForPod(String name) which checks that all Kafka pods are ready. 
In the previous realisation, this method just wait when pod will have status "Running". After improvement, the method waits when all pods will be ready.
P.S: Running != Ready for some system tests

### Checklist
- [x] Make sure all tests pass

### CODE REVIEW

File: common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java [hunk 1]
```diff
@@ -234,8 +234,20 @@ public K waitForDeployment(String name) {
 
     @Override
     public K waitForPod(String name) {
+        // wait when all pods are ready
         return waitFor("pod", name,
-            actualObj -> "Running".equals(actualObj.get("status").get("phase").asText()));
+            actualObj -> {
+                JsonNode containerStatuses = actualObj.get("status").get("containerStatuses");
+                ArrayList<Boolean> ready = new ArrayList<>();
+                if (containerStatuses.isArray()) {
```
Comment (tombentley) [inline common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java]: If `containerStatuses` is null, then NPE.

File: common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java [hunk 2]
```diff
@@ -234,8 +234,20 @@ public K waitForDeployment(String name) {
...
+                    for (final JsonNode objNode : containerStatuses) {
+                        ready.add(objNode.get("ready").asBoolean());
```
Comment (tombentley) [inline common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java]: No need for `ready`: If `if (!objNode.get("ready").asBoolean()) return false` and if you get all the way through the loop then `return true`.

File: common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java [hunk 3]
```diff
@@ -234,8 +234,22 @@ public K waitForDeployment(String name) {
...
+        // wait when all pods are ready
         return waitFor("pod", name,
-            actualObj -> "Running".equals(actualObj.get("status").get("phase").asText()));
+            actualObj -> {
+                try {
+                    JsonNode containerStatuses = actualObj.get("status").get("containerStatuses");
+                    if (containerStatuses.isArray()) {
+                        for (final JsonNode objNode : containerStatuses) {
+                            if (!objNode.get("ready").asBoolean()) return false;
+                        }
```
Comment (tombentley) [inline common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java]: The re`turn true` should be after the `for` loop, not after the `if` block

File: common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java [hunk 4]
```diff
@@ -234,8 +234,22 @@ public K waitForDeployment(String name) {
 
     @Override
     public K waitForPod(String name) {
+        // wait when all pods are ready
         return waitFor("pod", name,
-            actualObj -> "Running".equals(actualObj.get("status").get("phase").asText()));
+            actualObj -> {
+                try {
+                    JsonNode containerStatuses = actualObj.get("status").get("containerStatuses");
+                    if (containerStatuses.isArray()) {
```
Comment (tombentley) [inline common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java]: `if (containerStatuses != null && containerStatuses.isArray()) {` then you don't need the `catch (NPE)`

File: common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java [hunk 5]
```diff
@@ -234,8 +234,22 @@ public K waitForDeployment(String name) {
...
+                        for (final JsonNode objNode : containerStatuses) {
+                            if (!objNode.get("ready").asBoolean()) return false;
+                        }
+                    }
+                    return true;
```
Comment (tombentley) [inline common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java]: The return after the if block should be a `return false`

File: common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java (hunk 4 above)
Comment (serrss) [reply common-test/src/main/java/io/strimzi/test/k8s/BaseKubeClient.java]: The code looks match better without try/catch for NPE

**Comment by tombentley** [comment]
@serrss the cod enow looks good to me. Have you tested it?

**Comment by serrss** [comment]
@tombentley thanks for the review, I tested this code using current tests and also I debugged them to be sure that method waits that all pods are ready

**Pull request merged**